src/utils/notification/htmlBuilder.ts

```diff
@@ -0,0 +1,463 @@
+      });
+      subject = subject.replace('{{now.time}}', nowToString);
+    }
+    if (subject.includes('{{now.datetime}}')) {
+      const nowToString = new Date().toLocaleString();
+      subject = subject.replace('{{now.datetime}}', nowToString);
+    }
+
+    if (subject.includes('{{today.date}}')) {
+      const todayToString = new Date().toDateString();
+      subject = subject.replace('{{today.date}}', todayToString);
+    }
+  }
+  return subject;
+};
+
+/**
+ * Replaces macros in header with values
+ *
+ * @param header Header object returned from DB
+ * @param header.headerStyle Header body styles
+ * @param header.headerHtml Header HTML from TinyMCE
+ * @param header.headerHtmlStyle Header text styles
+ * @param header.headerLogo Header logo in base64 rep
+ * @param header.headerLogoStyle Header logo styles
+ * @param header.headerLogo.__zone_symbol__value Header logo in base64
+ * @returns mutated string with replaced macro
+ */
+export const replaceHeader = (header: {
+  headerStyle: string;
+  headerHtml: string;
+  headerHtmlStyle: string;
+  headerLogo: {
+    // eslint-disable-next-line @typescript-eslint/naming-convention
+    __zone_symbol__value: string;
+  };
+  headerLogoStyle: string;
+}): string => {
+  let headerString = `<table border="0" cellpadding="0" cellspacing="0" width="760px" style="padding: 10px;">
+  <tbody>
+  <tr>`;
+
+  if (header.headerLogo) {
+    headerString += `<td style="padding: 10px;">
+          <a href="" style="display: block; border-style: none !important; text-align: center; border: 0 !important;">
+              <img width="120" data-imagetype="DataUri"  src="cid:headerImage" style="padding: 10px;">
+          </a>
+      </td>
+    `;
+  }
+
+  if (header.headerHtml) {
+    const inthelastMatcher = new RegExp(
+      '{{((?!today.date|now.datetime|now.time)[^{{|^}}]+)}}',
+      'g'
+    );
+    const matches = header.headerHtml.matchAll(inthelastMatcher);
+    for (const match of matches) {
+      const splitToken = match[1].split('.');
+      const currentDate = new Date();
+      // Current date offset by minutes param
+      const pastDate = new Date(
+        currentDate.getTime() - Number(splitToken[2]) * 60000
+      );
+
+      // Past Date to date (mm/dd/yyyy)
+      const formattedPastDate = pastDate.toLocaleDateString('en-US', {
+        year: 'numeric',
+        month: 'long',
+        day: '2-digit',
+      });
+
+      // Past Date to time (hh:mm)
+      const formattedPastTime = pastDate.toLocaleTimeString('en-US', {
+        hour: '2-digit',
+        minute: '2-digit',
+      });
+
+      // Current Date to date (mm/dd/yyyy)
+      const formattedCurrentDate = currentDate.toLocaleDateString('en-US', {
+        year: 'numeric',
+        month: 'long',
+        day: '2-digit',
+      });
+
+      // Current Date to time (hh:mm)
+      const formattedCurrentTime = currentDate.toLocaleTimeString('en-US', {
+        hour: '2-digit',
+        minute: '2-digit',
+      });
```

Comment:
Could we make those localeFormats constants so they can be changed easily if required?